### PR TITLE
Implement Send Email Feature 

### LINKED ISSUE #34: Implement the admin dashboard - sent email view
### Description
As part of this issue, we need to create the admin dashboard - sent email view, which should closely match the provided mockup. This will provide the admins with a way to view all the sent emails.

### Tasks
1. Set up the basic React components as required.
1. Implement the layout and design of the admin dashboard - sent email view, following the provided mockup.
1. Integrate the necessary components and assets to achieve the desired functionality and visual appearance.
1. Ensure responsiveness and compatibility across different devices and screen sizes.

Mockup: https://www.figma.com/file/nehm2ff8aR7NkPerg39fK9/ScholarX-V2-%7C-Wireframes?type=design&node-id=1974-4052&mode=design&t=zYi191UOlImykhe8-4

image

Note: You may want to mock the API refer to the [API doc](https://documenter.getpostman.com/view/27421496/2s93m1a4ac#ccbab660-6313-428f-9063-c12d27b4606b) to see the relevant details.

### Acceptance Criteria
- The user interface closely resembles the provided mockup.
- All required sections, elements, and content are present and displayed correctly.
- The interface is responsive and works well on various devices and screen sizes.
- The implementation follows best practices and coding standards.
- Unit tests are written.

### PR DESCRIPTION
**Pull request by mayura-andrew**
## Purpose
The purpose of this PR is to fix #34 

## Goals
Implement simple UI for send emails.
View sent emails history
Connect with Email API
Implement logic to check whether the user has opened our sent email.

## Approach
Implement a separate email API to handle backend related works and connect with frontend.
Separate Email API repo : https://github.com/mayura-andrew/send-bulk-email-client-api 
## Screenshots
image
image

##  Checklist
- [ ] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

## Related PRs
Implemented #99 
## Learning
SMTP, Go Lang based API development

### CODE REVIEW

File: src/pages/Dashboard/scenes/Emails/Emails.tsx [hunk 1]
```diff
@@ -1,9 +1,298 @@
...
+                                break;
+                              case 'rejectedMentees':
+                                setSelectedMentees(rejectedMenteesEmails);
+                                setSelect('Mentees');
+                                break;
+                              default:
+                                setSelectedMentees([]);
+                            }
+                          }}
+                        >
```
Comment (anjula-sack) [inline src/pages/Dashboard/scenes/Emails/Emails.tsx]: move this to a separate function

File: src/pages/Dashboard/scenes/Emails/Emails.tsx [hunk 2]
```diff
@@ -1,9 +1,298 @@
...
+                                    fill="currentColor"
+                                  />
+                                  <path
+                                    d="M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z"
+                                    fill="currentFill"
+                                  />
+                                </svg>
```
Comment (anjula-sack) [inline src/pages/Dashboard/scenes/Emails/Emails.tsx]: Create components for the SVGs files under assets folder

File: src/pages/Dashboard/scenes/Emails/Emails.tsx [hunk 3]
```diff
@@ -1,9 +1,298 @@
...
+      .get(`${EMAILAPI_URL}/api/v1/sent`)
+      .then((response) => {
+        setEmails(response.data.recipient);
+      })
+      .catch((error) => {
+        console.error('Error:', error);
+      });
+  }, []);
+
+  console.log(EMAILAPI_URL);
```
Comment (anjula-sack) [inline src/pages/Dashboard/scenes/Emails/Emails.tsx]:
```suggestion
```

File: src/components/Dashboard/scenes/Emails/CheckButton.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
+// CheckButton.tsx
```
Comment (anjula-sack) [inline src/components/Dashboard/scenes/Emails/CheckButton.tsx]:
```suggestion
```

File: src/components/Dashboard/scenes/Emails/EmailHistory.tsx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+import { useQuery } from '@tanstack/react-query';
+import { EMAILAPI_URL } from '../../../../constants';
+
+interface Email {
+  recipients: string;
+  subject: string;
+  status: string;
+  opened: boolean;
+  sentTime: string;
+}
```
Comment (anjula-sack) [inline src/components/Dashboard/scenes/Emails/EmailHistory.tsx]: Move this to types.ts

File: src/components/Dashboard/scenes/Emails/EmailTemplate.tsx [hunk 6]
```diff
@@ -0,0 +1,184 @@
+// EmailTemplate.tsx
```
Comment (anjula-sack) [inline src/components/Dashboard/scenes/Emails/EmailTemplate.tsx]:
```suggestion
```

File: src/components/Dashboard/scenes/Emails/Emails.tsx [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+      .get(`${EMAILAPI_URL}/api/v1/sent`)
+      .then((response) => {
+        setEmails(response.data.recipient);
+      })
+      .catch((error) => {
+        console.error('Error:', error);
+      });
+  }, []);
+
+  console.log(EMAILAPI_URL);
```
Comment (anjula-sack) [inline src/components/Dashboard/scenes/Emails/Emails.tsx]:
```suggestion
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/components/Dashboard/scenes/Emails/Emails.tsx [hunk 8]
```diff
@@ -0,0 +1,300 @@
...
+      })
+      .finally(() => {
+        setTimeout(() => {
+          setIsLoading(false);
+        }, 1000);
+      });
+  };
+  useEffect(() => {
+    checkApiStatus();
+  }, []);
```
Comment (anjula-sack) [inline src/components/Dashboard/scenes/Emails/Emails.tsx]: We don't need the health check. If the server is down we cannot send the emails anyway.
```suggestion
```

File: src/components/Dashboard/scenes/Emails/Emails.tsx [hunk 9]
```diff
@@ -0,0 +1,300 @@
...
+
+  const handleFormSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const emailData = {
+      sender: 'mayuraalahakoon@gmail.com',
```
Comment (anjula-sack) [inline src/components/Dashboard/scenes/Emails/Emails.tsx]:
```suggestion
      sender: 'mayuraalahakoon@gmail.com',
```
Move this to the constants.

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/components/Dashboard/scenes/Emails/Emails.tsx (hunk 9 above)
Comment (mayura-andrew) [reply src/components/Dashboard/scenes/Emails/Emails.tsx]: i'll update this

**Review by anjula-sack — CHANGES_REQUESTED** [review]
Why do we have duplicated components?

File: src/pages/Dashboard/scenes/Emails/EmailHistory.tsx [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+
+  const fetchEmails = useCallback(() => {
+    setIsLoading(true);
+    setTimeout(() => {
+      axios
+        .get(`${EMAILAPI_URL}/api/v1/sent`)
+        .then((response) => {
+          setEmails(
+            Array.isArray(response.data.emails) ? response.data.emails : []
+          );
```
Comment (anjula-sack) [inline src/pages/Dashboard/scenes/Emails/EmailHistory.tsx]: The API should send [] if there's not data

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/pages/Dashboard/scenes/Emails/EmailHistory.tsx (hunk 10 above)
Comment (mayura-andrew) [reply src/pages/Dashboard/scenes/Emails/EmailHistory.tsx]: like this  ?
image

Comment (anjula-sack) [reply src/pages/Dashboard/scenes/Emails/EmailHistory.tsx]: yeah

File: example.env [hunk 11]
```diff
@@ -1 +1,3 @@
 VITE_API_URL=http://localhost:3000/api
+VITE_EMAILAPI_URL=https://64.227.135.79/
+VITE_SENDER_EMAIL=
```
Comment (anjula-sack) [inline example.env]: EOL
```suggestion
VITE_SENDER_EMAIL=

```

File: src/components/Dashboard/scenes/Emails/EmailHistory.tsx [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+        })
+        .catch((error) => {
+          console.error('Error:', error);
+          setIsLoading(false);
+        });
+    }, 1000);
+  }, []);
+  useEffect(() => {
+    fetchEmails();
+  }, [fetchEmails, refreshCount]);
```
Comment (anjula-sack) [inline src/components/Dashboard/scenes/Emails/EmailHistory.tsx]: no need to fetch emails always

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/components/Dashboard/scenes/Emails/EmailHistory.tsx (hunk 12 above)
Comment (mayura-andrew) [reply src/components/Dashboard/scenes/Emails/EmailHistory.tsx]: ok i'll remove this and update

File: src/components/Dashboard/scenes/Emails/EmailHistory.tsx [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+        .get(`${EMAILAPI_URL}/api/v1/sent`)
+        .then((response) => {
+          setEmails(response.data.emails);
+          setIsLoading(false);
+        })
+        .catch((error) => {
+          console.error('Error:', error);
+          setIsLoading(false);
+        });
+    }, 1000);
```
Comment (anjula-sack) [inline src/components/Dashboard/scenes/Emails/EmailHistory.tsx]: Remove the timeout. Use a hook to sent mails

File: src/components/Dashboard/scenes/Emails/EmailTemplate.tsx [hunk 14]
```diff
@@ -0,0 +1,180 @@
...
```
Comment (anjula-sack) [inline src/components/Dashboard/scenes/Emails/EmailTemplate.tsx]:
```suggestion
```

File: src/components/RegisterModal/index.tsx [hunk 15]
```diff
@@ -153,6 +154,7 @@ const RegisterModal: React.FC<RegisterModalProps> = ({ handleClose }) => {
                 >
                   Register
                 </button>
+                <GoogleBtn />
```
Comment (anjula-sack) [inline src/components/RegisterModal/index.tsx]:
```suggestion
                <GoogleLoginButton />
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/components/Dashboard/scenes/Emails/EmailHistory.tsx (hunk 13 above)
Comment (mayura-andrew) [reply src/components/Dashboard/scenes/Emails/EmailHistory.tsx:24]:
```
const fetchEmails = useCallback(() => {
    setIsLoading(true);
    axios
        .get(`${EMAILAPI_URL}/api/v1/sent`)
        .then((response) => {
          setEmails(response.data.emails);
          setIsLoading(false);
        })
        .catch((error) => {
          console.error('Error:', error);
          setIsLoading(false);
        });
  }, []);

  useEffect(() => {
    fetchEmails();
  }, [fetchEmails]);
```

like this ?

File: src/hooks/useFetchEmails.ts [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+import axios, { type AxiosError } from 'axios';
+import { EMAILAPI_URL } from '../constants';
+import { type Email } from '../types';
+
+const fetchEmails: QueryFunction<Email[], QueryKey> = async () => {
+  const response = await axios.get(`${EMAILAPI_URL}/api/v1/sent`);
+  return response.data.emails;
+};
+
+export const useFetchEmails = () => {
```
Comment (anjula-sack) [inline src/hooks/useFetchEmails.ts]:
```suggestion
export const useEmails = () => {
```

File: src/assets/svg/LoadingSmallSVG.tsx [hunk 17]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+const LoadingSmallSVG: React.FC = () => (
```
Comment (anjula-sack) [inline src/assets/svg/LoadingSmallSVG.tsx]: Just use the same loading icon and pass the width height as props

File: src/assets/svg/LoadingSVG.tsx [hunk 18]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+const LoadingSVG: React.FC = () => (
```
Comment (anjula-sack) [inline src/assets/svg/LoadingSVG.tsx]:
```suggestion
const Loading: React.FC = () => (
```

File: src/components/Dashboard/scenes/Emails/EmailTemplate.tsx [hunk 19]
```diff
@@ -0,0 +1,179 @@
+import React from 'react';
+
+interface EmailTemplateProps {
+  subject: string;
+  body: string;
+  recipient: string;
```
Comment (anjula-sack) [inline src/components/Dashboard/scenes/Emails/EmailTemplate.tsx]: Do we need it?

File: src/pages/Dashboard/scenes/Emails/Emails.tsx [hunk 20]
```diff
@@ -1,9 +1,224 @@
...
+    }
+  };
+
+  useEffect(() => {
+    if (message.length > 0) {
+      const timer = setTimeout(() => {
+        setMessage('');
+      }, 1000);
+      return () => {
+        clearTimeout(timer);
```
Comment (anjula-sack) [inline src/pages/Dashboard/scenes/Emails/Emails.tsx]: Why do you need a timeout?

File: src/pages/Dashboard/scenes/Emails/Emails.tsx [hunk 21]
```diff
@@ -1,9 +1,224 @@
...
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(emailData),
+      });
+
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.statusText}`);
+      }
```
Comment (anjula-sack) [inline src/pages/Dashboard/scenes/Emails/Emails.tsx]: Use a mutation here from react query

File: src/pages/Dashboard/scenes/Emails/Emails.tsx [hunk 22]
```diff
@@ -1,9 +1,224 @@
...
+      setMessage('Success: Emails sent!');
+    } catch (error) {
+      setMessage(`Error : ${(error as Error).message}`);
+    } finally {
+      setIsLoading(false);
+      setTimeout(() => {
+        setIsLoading(false);
+      }, 1000);
```
Comment (anjula-sack) [inline src/pages/Dashboard/scenes/Emails/Emails.tsx]:
```suggestion
```

File: src/pages/Dashboard/scenes/Emails/Emails.tsx [hunk 23]
```diff
@@ -1,9 +1,224 @@
...
+      return () => {
+        clearTimeout(timer);
+      };
+    }
+  }, [message]);
+
+  const handleSelectChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    switch (e.target.value) {
+      case 'allMentors':
+        setSelectedMentees(allMentorsEmails);
```
Comment (anjula-sack) [inline src/pages/Dashboard/scenes/Emails/Emails.tsx]: This is supposed to be a string. Check the other lint errors as well

File: src/pages/Dashboard/scenes/Emails/Emails.tsx [hunk 24]
```diff
@@ -1,9 +1,224 @@
...
+                    recipient={select}
+                  />
+                </div>
+              </div>
+            </div>
+          </>
+        ) : (
+          <>
+            <EmailHistory />
+          </>
```
Comment (anjula-sack) [inline src/pages/Dashboard/scenes/Emails/Emails.tsx]:
```suggestion
            <EmailHistory />
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/pages/Dashboard/scenes/Emails/Emails.tsx (hunk 21 above)
Comment (mayura-andrew) [reply src/pages/Dashboard/scenes/Emails/Emails.tsx]: ok i'll update this

File: src/components/Dashboard/scenes/Emails/EmailTemplate.tsx (hunk 19 above)
Comment (mayura-andrew) [reply src/components/Dashboard/scenes/Emails/EmailTemplate.tsx]: I removed this Email Template Component. Is it necessary to display, no right ?

Comment (mayura-andrew) [reply src/components/Dashboard/scenes/Emails/EmailTemplate.tsx]: image
Updated

File: src/pages/Dashboard/scenes/Emails/Emails.tsx (hunk 24 above)
Comment (mayura-andrew) [reply src/pages/Dashboard/scenes/Emails/Emails.tsx]: @anjula-sack  in history component, I'm just rendering each email's sender email, subject, status of sent,  open status, and sent time. so when I sent test emails with long subject text, that table quite getting big row each emails. 
image

File: src/components/Dashboard/scenes/Emails/EmailTemplate.tsx (hunk 19 above)
Comment (mayura-andrew) [reply src/components/Dashboard/scenes/Emails/EmailTemplate.tsx]: I can remove this Preview component or, instead,  I tried this also 
[Screencast from 2024-05-21 05-50-31.webm](https://github.com/sef-global/scholarx-frontend/assets/48531182/6f6a5ffc-56c9-4361-9690-f33208148c80)

File: src/pages/Dashboard/scenes/Emails/Emails.tsx (hunk 20 above)
Comment (mayura-andrew) [reply src/pages/Dashboard/scenes/Emails/Emails.tsx]: I removed it.

File: src/pages/Dashboard/scenes/Emails/Emails.tsx (hunk 23 above)
Comment (mayura-andrew) [reply src/pages/Dashboard/scenes/Emails/Emails.tsx]: still I could not implement this. I couldn't figure out how the API call should be to call and fetch specific emails from backend!

File: src/components/Dashboard/scenes/Emails/EmailTemplate.tsx (hunk 19 above)
Comment (anjula-sack) [reply src/components/Dashboard/scenes/Emails/EmailTemplate.tsx]: Let's remove the preview for now.

File: src/pages/Dashboard/scenes/Emails/Emails.tsx (hunk 23 above)
Comment (anjula-sack) [reply src/pages/Dashboard/scenes/Emails/Emails.tsx]: for now, console.log the value. I will integrate it.

File: src/pages/Dashboard/scenes/Emails/Emails.tsx (hunk 24 above)
Comment (anjula-sack) [reply src/pages/Dashboard/scenes/Emails/Emails.tsx]: Truncate the subject using taiilwind css

File: src/components/Dashboard/scenes/Emails/EmailTemplate.tsx (hunk 19 above)
Comment (mayura-andrew) [reply src/components/Dashboard/scenes/Emails/EmailTemplate.tsx]: ok

File: src/pages/Dashboard/scenes/Emails/Emails.tsx (hunk 24 above)
Comment (mayura-andrew) [reply src/pages/Dashboard/scenes/Emails/Emails.tsx]: okay I'll do it

File: src/pages/Dashboard/scenes/Emails/Emails.tsx (hunk 21 above)
Comment (mayura-andrew) [reply src/pages/Dashboard/scenes/Emails/Emails.tsx]: @anjula-sack, I'm working on this query mutation. Can you look at what I've implemented? It is not complete.😞

File: src/pages/Dashboard/scenes/Emails/Emails.tsx [hunk 25]
```diff
@@ -1,9 +1,198 @@
...
+import { useSendEmail } from '../../../../hooks/useEmails';
+import { type EmailData } from '../../../../types';
 
 const Emails: React.FC = () => {
+  const [selectedMentees, setSelectedMentees] = useState<string[]>([]);
+  const [view, setView] = useState('sent');
+  const [select, setSelect] = useState('');
+  const [message, setMessage] = useState('');
+  const sendEmailMutation = useSendEmail();
+
```
Comment (mayura-andrew) [inline src/pages/Dashboard/scenes/Emails/Emails.tsx:21]: is this correct way to implement this query mutation ?

File: src/hooks/useEmails.ts [hunk 26]
```diff
@@ -0,0 +1,45 @@
...
+};
+
+const useSendEmail = () => {
+  const mutation = useMutation({
+    mutationFn: sendEmail,
+  });
+
+  return mutation;
+};
+
```
Comment (mayura-andrew) [inline src/hooks/useEmails.ts]: here also

File: src/pages/Dashboard/scenes/Emails/Emails.tsx [hunk 27]
```diff
@@ -1,9 +1,198 @@
...
+    e.preventDefault();
+    sendEmailMutation.mutate(formData, {
+      onSuccess: (data) => {
+        console.log('Email sent successfully:', data);
+      },
+      onError: (error) => {
+        console.error('Error sending email:', error);
+      },
+    });
+  };
```
Comment (mayura-andrew) [inline src/pages/Dashboard/scenes/Emails/Emails.tsx]: this function use that sendEmailMutation

File: src/components/Dashboard/scenes/Emails/EmailHistory.tsx [hunk 28]
```diff
@@ -0,0 +1,85 @@
...
+import Loading from '../../../../assets/svg/Loading';
+import { useEmails } from '../../../../hooks/useEmails';
+
+function truncateSubject(subject: string, wordLimit: number) {
+  const words = subject.split(' ');
+  if (words.length > wordLimit) {
+    return words.slice(0, wordLimit).join(' ') + '...';
+  }
+  return subject;
+}
```
Comment (anjula-sack) [inline src/components/Dashboard/scenes/Emails/EmailHistory.tsx]: You don't need to use a function for this. Just use the truncate class https://tailwindcss.com/docs/text-overflow

File: src/pages/Dashboard/scenes/Emails/Emails.tsx [hunk 29]
```diff
@@ -1,9 +1,197 @@
...
+  ) => {
+    const { value } = e.target;
+    setFormData((prevData) => ({
+      ...prevData,
+      recipients: value.split(','),
+    }));
+  };
+
+  const handleSelectChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    console.log(`Selected: ${select}`);
```
Comment (anjula-sack) [inline src/pages/Dashboard/scenes/Emails/Emails.tsx]:
```suggestion
```

File: src/hooks/useEmails.ts [hunk 30]
```diff
@@ -0,0 +1,46 @@
...
+  const response = await axios.get(`${EMAILAPI_URL}/api/v1/sent`);
+  return response.data.emails;
+};
+
+const sendEmail = async (emailData: EmailData): Promise<SendEmailsResponse> => {
+  const response = await axios.post(`${EMAILAPI_URL}/api/v1/send`, emailData);
+  return response.data;
+};
+
+const useSendEmail = () => {
```
Comment (anjula-sack) [inline src/hooks/useEmails.ts]: Add this inside the useEmails hook itself, go through this https://github.com/sef-global/scholarx-frontend/blob/main/src/hooks/admin/useMentees.ts

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/hooks/useEmails.ts [hunk 31]
```diff
@@ -0,0 +1,45 @@
...
+  return response.data.emails;
+};
+
+const sendEmail = async (emailData: EmailData): Promise<SendEmailsResponse> => {
+  const response = await axios.post(`${EMAILAPI_URL}/api/v1/send`, emailData);
+  return response.data;
+};
+
+export const useEmails = () => {
+  const useSendEmail = () => {
```
Comment (anjula-sack) [inline src/hooks/useEmails.ts]: Use prefix is used for hooks this is not a hook. 
```suggestion
  const sendEmail = () => {
```

File: src/hooks/useEmails.ts [hunk 32]
```diff
@@ -0,0 +1,45 @@
...
+    const mutation = useMutation({
+      mutationFn: sendEmail,
```
Comment (anjula-sack) [inline src/hooks/useEmails.ts]:
```suggestion
      mutationFn: async (emailData: EmailData): Promise<SendEmailsResponse> => {
  const response = await axios.post(`${EMAILAPI_URL}/api/v1/send`, emailData);
  return response.data;
},
```

File: src/hooks/useEmails.ts [hunk 33]
```diff
@@ -0,0 +1,45 @@
...
+    queryFn: fetchEmails,
+  });
+
+  const mutation = useSendEmail();
+
+  return {
+    isLoading,
+    error,
+    data,
+    mutation,
```
Comment (anjula-sack) [inline src/hooks/useEmails.ts]:
```suggestion
    sendEmails,
```

File: src/hooks/useEmails.ts [hunk 34]
```diff
@@ -0,0 +1,45 @@
...
+    isPending: mutation.status === 'pending',
+    responseMessage: mutation.data?.message,
```
Comment (anjula-sack) [inline src/hooks/useEmails.ts]:
```suggestion
```

File: src/pages/Dashboard/scenes/Emails/Emails.tsx [hunk 35]
```diff
@@ -1,9 +1,189 @@
...
+import EmailHistory from '../../../../components/Dashboard/scenes/Emails/EmailHistory';
+import { EMAILAPI_SENDER } from '../../../../constants';
+import Loading from '../../../../assets/svg/Loading';
+import { useEmails } from '../../../../hooks/useEmails';
+import { type EmailData } from '../../../../types';
 
 const Emails: React.FC = () => {
+  const [view, setView] = useState('sent');
+  const [message, setMessage] = useState('');
+  const sendEmailMutation = useEmails();
```
Comment (anjula-sack) [inline src/pages/Dashboard/scenes/Emails/Emails.tsx]:
```suggestion
  const { sendEmails } = useEmails();
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/hooks/useEmails.ts [hunk 36]
```diff
@@ -0,0 +1,53 @@
...
+  type QueryKey,
+  useMutation,
+} from '@tanstack/react-query';
+import axios, { type AxiosError } from 'axios';
+import { EMAILAPI_URL } from '../constants';
+import { type Email, type EmailData, type SendEmailsResponse } from '../types';
+
+const fetchEmails: QueryFunction<Email[], QueryKey> = async () => {
+  const response = await axios.get(`${EMAILAPI_URL}/api/v1/sent`);
+  console.log(response.data.emails);
```
Comment (anjula-sack) [inline src/hooks/useEmails.ts]:
```suggestion
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: package.json [hunk 37]
```diff
@@ -22,6 +22,7 @@
     "@tanstack/react-query-devtools": "^5.37.1",
     "axios": "^1.4.0",
     "dotenv": "^16.0.3",
+    "esbuild": "^0.21.4",
```
Comment (anjula-sack) [inline package.json]: What is this package? @mayura-andrew

Comment (mayura-andrew) [reply package.json]: When i tried  to run ```npm run dev``` in my local but i got a error  that said the package "@esbuild/linux-x64" could not be found, which is needed by esbuild.  then i install ```npm install esbuild```.

**Review by anjula-sack — APPROVED** [review]
Good job! @mayura-andrew

**Pull request merged**